subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/AbstractJavaCompileSpecFactory.java

```diff
@@ -39,6 +39,7 @@ public T create() {
         if (toolchain != null) {
```

Comment:
Is it possible that the toolchain is `null` here?

Reply:
Yes, when we arrive here from `GroovyCompile`